### PR TITLE
K8s Gateway object creation in Wizard

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/5502

Screenshot from 2022-10-10 12-58-35
Screenshot from 2022-10-10 12-57-22

For testing:
Go to Istio Config list page.
"Actions" dropdown, added new option "Create K8s Gateway".
When Gateway API is enabled (https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/ also add 'app' label into Gateway object), newly added option is enabled. What Gateway API CRD is not installed, then option is disabled.

### CODE REVIEW

**Comment by hhovsepy** [comment]
The K8s Gateway API "Gateway" object itself contains more fields, but not all property's fields are touched in a scope of this PR. Further tasks will be created based on community experience and needs in next sprints.

**Comment by josunect** [comment]
Hi, @hhovsepy 

Following the steps to create a k8s gateway, it creates 2 objects: A gateway and a k8s GW. This is expected? 

image

**Comment by hhovsepy** [comment]
@josunect actually the second one is auto-generated by Istio, for internal use. When the Istio configs switched to load resources from Istio Registry, all Istio auto-generated objects are loaded, and we leave them as it is not to hide from users.

**Comment by josunect** [comment]
> @josunect actually the second one is auto-generated by Istio, for internal use. When the Istio configs switched to load resources from Istio Registry, all Istio auto-generated objects are loaded, and we leave them as it is not to hide from users.

Ah! I remember. Thanks!

**Review by josunect — COMMENTED** [review]
Tested Create new GW: 
image

And removed: 
image


Tested k8sGW is disabled: 

image


https://github.com/kiali/kiali/pull/5537/files#diff-163a601f65fbd474e86f1928ac31ba72659f4fd009c067c6f3226ce066f8861fR1198

There's no need to add newK8sNetworkingConfigType len also? 

`allRP := make(models.ResourcesPermissions, len(newNetworkingConfigTypes)+len(newSecurityConfigTypes)+len(newK8sNetworkingConfigTypes))`

This is just a suggestion, because this is similar in other objects. When I first seen this page, it took me a while to understand the Listener and Address fields. So, I would suggest to improve these fields to something like: 

image

**Review by jshaughn — COMMENTED** [review]
Should this new support have any additional cypress coverage?

**Comment by hhovsepy** [comment]
> Should this new support have any additional cypress coverage?

It requires Gateway CRD to be installed in cluster, otherwise cypress can check that menu action item is disabled.

File: frontend/src/pages/IstioConfigNew/GatewayForm/AddressBuilder.tsx
Comment (jshaughn) [inline frontend/src/pages/IstioConfigNew/GatewayForm/AddressBuilder.tsx:150]:
```suggestion
        {!this.canAddAddress() && <span className={warningStyle}>An Address needs Type and Value sections defined</span>}
```

**Review by jshaughn — COMMENTED** [review]
Tested and working, I have nothing to add to the good review by @josunect.  I made one trivial grammar suggestion to the warning message.  I will approve when the cypress change is added.

**Comment by josunect** [comment]
> @josunect seems all comments approached, thank you! Screenshot from 2022-10-13 16-56-21

Thanks, @hhovsepy ! All the changes seems to be addressed. 
I will leave it pending for the changes suggested by @jshaughn

**Comment by hhovsepy** [comment]
@josunect @jshaughn  have added a cypress test

**Review by jshaughn — APPROVED** [review]
LG, thanks for adding test.  Approving.  If CI passes feel free to merge for this release.

**Pull request merged**
